superset/assets/javascripts/SqlLab/components/QueryAutoRefresh.jsx

```diff
@@ -14,6 +14,17 @@ class QueryAutoRefresh extends React.PureComponent {
   componentWillUnmount() {
     this.stopTimer();
   }
+  shouldCheckForQueries() {
+    // if there are started or pending queries, this method should return true
+    const { queries } = this.props;
+    const queryKeys  = Object.keys(queries);
+    const pendingOrStartedQueries = queryKeys.map((key) => {
```

Comment:
a `forEach` with `return true` when the condition is met would exit faster in many cases, you'll need a `return false` after the forEach block. The combo of `map` here and filter at line 26 is not optimal and a bit hard to read.